Created Issue and Pull Request Templates

jflopezfernandez opened the pull request:
### Fixes #42 
In addition, I moved the CONTRIBUTING.md file to the .github
directory, as it will still be displayed correctly, [per the
github blog](https://github.blog/2016-02-17-issue-and-pull-request-templates/), but it reduces the clutter on the main
project page.

ranolfi commented:
It seems to me that keeping CONTRIBUTING.md in the project root is standard practice - it doesn't necessarily pertain to GitHub alone. Can we move it back?

jflopezfernandez commented:
Sure, I'll push the change in a sec

ranolfi commented:
Thanks. It would've been best to amend the previous commit and force-push, but don't worry now - this is so small and simple that at this time I prefer to simply squash to master instead.

ranolfi commented:
Inviting Paul and Mike to review, but if someone else wants to step in feel free to do so. To be merged soon after.

luap42 commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,6 @@
+
```
Not sure, whether this is standard practice, but I'd add something like this to the beginning:

    ### Pull Request summary
    
    (Describe what features/behaviors your Pull Request adds/changes/removes)
    
    ### Link to Project Card/Issue
    
    (Add a link to the Project Card or Issue you're working at.)

jflopezfernandez replied:
Nice, I'll go ahead and add that in

luap42 commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,2 @@
+
+> Please describe the issue.
```
Maybe that's just me, but I am not sure how useful a file is that'll put "Please describe the issue" into the issue explanation box.

jflopezfernandez replied:
Dude I actually laughed out loud when I read your comment, because I was thinking the **exact** same thing to myself as I was writing it. My thought was that we might collaboratively come up with a more useful issue template, though. I'll see if I can find any resources to improve on it, and if not I'll just delete the file.

ranolfi replied:
Valid point from @luap42 - but I also approach this as a placeholder file to be improved later.

jflopezfernandez replied:
It actually ended up being a non-issue, because it was a legacy method anyways. The suggested method is to create a directory within the new `.github` directory called `ISSUE_TEMPLATES` that you populate with issue templates for bugs, security issues, etc. I just removed the issue template for now while I do some more research on the methodology, but I added links to the documentation in the commit message.

jflopezfernandez commented:
@ranolfi Since @luap42 requested more significant changes to the PR, do you want me to just go ahead and ammend the original commit myself?

ranolfi commented:
@jflopezfernandez if you can manage, and don't mind doing so, it'd be great :)

jflopezfernandez committed:
Created Pull Request Template

The original pull request also included an issue template,
but while I did find some resources with good template ideas,
the Github documentation states that single issue templates
are better handled by a new method, which I need to research
more to implement well enough to include in a pull request.

The official documentation for issue templates can be found
[here](https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser).

Additional documentation on Github pull request templates can
be found
[here](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository).

jflopezfernandez commented:
@ranolfi I reset to the base and force pushed so the pr is a single commit. Is this how you want these to be done?

ranolfi commented:
This is excellent, thanks. Just need a second approval for merging.
